common/rpc/middleware.go

```diff
@@ -115,6 +116,28 @@ func (m *InboundMetricsMiddleware) Handle(ctx context.Context, req *transport.Re
 	return h.Handle(ctx, req, resw)
 }
 
+// YarpcKey is the const for yarpc key
+const ComparatorYarpcKey = "cadence-visibility-override"
+
+// PinotComparatorMiddleware checks the header of a grpc request, and then override the context accordingly
+// note: for Pinot Migration only (Jan. 2024)
+type PinotComparatorMiddleware struct{}
+
+func (m *PinotComparatorMiddleware) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter, h transport.UnaryHandler) error {
+	yarpcKey, _ := req.Headers.Get(ComparatorYarpcKey)
+	if yarpcKey == persistence.Primary {
+		ctx = ContextWithOverride(ctx, persistence.Primary)
+	} else if yarpcKey == persistence.Secondary {
+		ctx = ContextWithOverride(ctx, persistence.Secondary)
+	}
+	return h.Handle(ctx, req, resw)
+}
+
+// ContextWithOverride adds a value in ctx
+func ContextWithOverride(ctx context.Context, value string) context.Context {
```

Comment:
this just needs to use a name which indicates it's specifically for the Visibility override. Consider a name like `contextWithVisibilityOverride`